src/constraintFunctions.f90

```diff
@@ -219,15 +219,15 @@ subroutine getEnvVarsAtT( t )
     ! others (e.g., pressure, temperature and RH before H2O, and DEC
     ! before JFAC). Currently, this relies on
     ! environmentVariables.config having exactly the lines relating to
-    ! these 10 variables.
+    ! these variables.
     !
     ! To add another environment variable, the user would need to add
     ! that line to environmentVariables.config, and then add this as
-    ! element 11 in the orderedEnvVarNames initialisation below.  Its
+    ! element 12 in the orderedEnvVarNames initialisation below.  Its
     ! treatment needs defining in each of cases 1-3 and default below.
 
-    if ( size( envVarNames ) /= 10 ) then
-      write(stderr,*) 'size( envVarNames ) /= 10 in getEnvVarsAtT().'
+    if ( size( envVarNames ) /= 11 ) then
+      write(stderr,*) 'size( envVarNames ) /= 11 in getEnvVarsAtT().'
```

Comment:
I keep getting an error about the size of `orderedEnvVarNames`. This is the same as the size of `envVarNames` but I can't find where the size of  `envVarNames`  is set.

Reply:
`envVarNames` is allocated at https://github.com/AtChem/AtChem2/blob/e18a0341a11b4a9f7d57e46c2fd0a7c01cc18d85/src/inputFunctions.f90#L794 , which is based on  the number of lines in the .config file at https://github.com/AtChem/AtChem2/blob/e18a0341a11b4a9f7d57e46c2fd0a7c01cc18d85/src/inputFunctions.f90#L788 (read the comment for more info).

Reply:
I guess we're going to run into trouble here because the new `ASA` is optional. In the existing setup, while the order of env vars doesn't matter, they all must be present in `environmentVariables.config`. So in the current setup, you'd need to add a line for `ASA` to each of the `environmentVariables.config` files for each of the tests. That of course is not a very robust way of doing things.

We'd likely want to move to a position where `ASA` is optional - that would need some work to make the existing processing more flexible.

I think that's perfectly possible, but will need some work - this current setup is largely a legacy of the direct porting from the original.

Reply:
Okay, it seems that this would be the objective of issue #17, whenever we get to it.